### PR TITLE
Check If N and Its Double Exist #20

### LINKED ISSUE #20: Check If N and Its Double Exist
# DSA
Given an array arr of integers, check if there exist two indices i and j such that :

i != j
0 <= i, j < arr.length
arr[i] == 2 * arr[j]
 

Example 1:

Input: arr = [10,2,5,3]
Output: true
Explanation: For i = 0 and j = 2, arr[i] == 10 == 2 * 5 == 2 * arr[j]
Example 2:

Input: arr = [3,1,7,11]
Output: false
Explanation: There is no i and j that satisfy the conditions.
 

Constraints:

1.  2 <= arr.length <= 500
2.  -103 <= arr[i] <= 103

### PR DESCRIPTION
**Pull request by Prakul06**
Resolved Issue #20

### CODE REVIEW

File: misc/N and its Double.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
+int main()
+{
+    int N,a,i,j,count=0;
+    printf("Enter N=");
```
Comment (ken1000minus7) [inline misc/N and its Double.c]: In questions like these we do not give input prompt asking for input, so u can remove all the input prompts everywhere and assume that the user knows what has to be given

File: misc/N and its Double.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    scanf("%d",&N);
+    if(N>=2 & N<=500)
```
Comment (ken1000minus7) [inline misc/N and its Double.c]: As I mentioned above, u can assume that `N` lies in the range given in constraints so u do not need this condition

File: misc/N and its Double.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            printf("Enter Value at A[%d]=",a);
+            scanf("%d",&A[a]);
+            if(A[a]<=103 & A[a]>=-103)
+            i++;
+            else
+            {
+                printf("Invalid Input");
+                exit(0);
+            }
```
Comment (ken1000minus7) [inline misc/N and its Double.c]: Similarly u can remove these else conditions as well

File: misc/N and its Double.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        for(i=0;i<N;i++)
+        {
+            for(j=0;j<N;j++)
+            {
+                if(i!=j)
+                {
+                    if(A[i]==2*A[j])
+                    {
+                        printf("True");
```
Comment (ken1000minus7) [inline misc/N and its Double.c]: You should prefer ending your ans with a new line character i.e. `\n`, make this change everywhere

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]
Hey @Prakul06 your code is correct but there r a few things I would like to mention, in a question constraints refer to the range in which input will be given, so actually u do not need to check if the input follows the constraints, u can assume that the input will be in the given constraints, as u will start solving dsa problems you'll see that based on constraints, our solution can change as for example if the constraints say that the input can be very large, it means that we have to think of an optimized solution 

The code you have written is correct but there also exists a more optimized approach which u can look up if u want to but for now just take care of the points I will be mentioning below

**Comment by Prakul06** [comment]
Changes Done!!

**Comment by ken1000minus7** [comment]
You haven't taken input for n, and the input prompt for array elements is still there, when I said you need to remove the input prompt i meant the message asking for input and not the input itself

**Review by ken1000minus7 — APPROVED** [review]
Lgtm

**Pull request merged**